stumpy/gpu_stump.py

```diff
@@ -838,4 +850,4 @@ def gpu_stump(
 
     core._check_P(out[:, 0])
 
-    return out
+    return mparray(out, m=m, k=k)
```

Comment:
Yes, this is me planning for the future. Considering that a matrix profile is always dependent on `m` and `k`, I figured that it would be a good idea to always store that information along with the `mparray` and, maybe in the future, we can make `m` and `k` optional when computing other quantities. Come to think of it, I wonder if we should store the exclusion zone in `mparray` as well?!

Reply:
I am thinking that we use `mp.m_` or `mp._excl_zone_denom` as ways to check that things are "correct" but still require that the user pass them into functions like `stumpy.motifs`

Reply:
> Yes, this is me planning for the future.  Considering that a matrix profile is always dependent on m and k, I figured that it would be a good idea to always store that information along with the mparray

Got it. Cool!

> Come to think of it, I wonder if we should store the exclusion zone in mparray as well?!

Right! This is great actually! I noticed it was taken care of in your recent commits.